spec-tests/src/test/java/org/candlepin/spec/pools/PoolSpecTest.java

```diff
@@ -462,69 +460,64 @@ public void shouldIncludeCalculatedAttributesWhenFetchingPools() {
             .containsEntry("compliance_type", "Standard");
     }
 
-    // @Test
-    // public void shouldAllowFetchingCDNByPoolId() {
-    //     // Impl note:
-    //     // This test is currently blocked on a limitation with CDNs in that we can perform CRUD
-    //     // operations on them (as they are first-class objects for some reason), but the only way
-    //     // to assign a CDN to a pool is via manifest import or pool refresh -- neither of which are
-    //     // implementated at the time of writing.
-
-    //     ApiClient adminClient = ApiClients.admin();
-    //     OwnerDTO owner = this.createOwner(adminClient);
+    @Test
+    @OnlyInHosted
+    public void shouldAllowFetchingCDNByPoolId() {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = this.createOwner(adminClient);
+        HostedTestApi upstreamClient = adminClient.hosted();
 
-    //     CdnDTO cdn = new CdnDTO()
-    //         .name(StringUtil.randomSuffix("test_cdn"))
-    //         .label("Test CDN")
-    //         .url("https://cdn.test.com");
+        CdnDTO cdn = new CdnDTO()
+            .name(StringUtil.random("test_cdn"))
+            .label(StringUtil.random("Test CDN"))
+            .url("https://cdn.test.com");
 
-    //     cdn = adminClient.cdns().createCdn(cdn);
+        cdn = adminClient.cdns().createCdn(cdn);
 
-    //     ProductDTO product = Products.randomSKU()
-    //         .addAttributesItem(this.buildAttribute("virt_limit", "10"));
+        ProductDTO product = Products.randomSKU()
+            .addAttributesItem(this.buildAttribute("virt_limit", "10"));
 
-    //     product = adminClient.ownerProducts()
-    //         .createProductByOwner(owner.getKey(), product);
+        product = adminClient.ownerProducts()
+            .createProductByOwner(owner.getKey(), product);
 
-    //     PoolDTO pool = Pools.randomUpstream(product)
-    //         .setCdn(cdn);
+        upstreamClient.createProduct(product);
+        upstreamClient.createSubscription(Subscriptions.random(owner, product).cdn(cdn));
 
-    //     pool = adminClient.owners()
-    //         .createPool(owner.getKey(), pool);
+        AsyncJobStatusDTO job =  adminClient.owners().refreshPools(owner.getKey(), false);
+        adminClient.jobs().waitForJob(job.getId());
```

Comment:
I would verify that the refresh pools jobs was successful by asserting the AsyncJobStatusDTO state equals "FINISHED".

Reply:
done